### PR TITLE
prow: add snyk_token ExternalSecret

### PR DESCRIPTION
**Pull request by spiffxp**
Supercedes https://github.com/kubernetes/test-infra/pull/22298

Secret used by job in https://github.com/kubernetes/test-infra/pull/22293

### CODE REVIEW

File: infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods-externalsecrets.yaml [hunk 1]
```diff
@@ -8,12 +8,20 @@ metadata:
...
+---
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: snyk_token
+  namespace: test-pods
+spec:
+  backendType: gcpSecretsManager
+  data:
+  - key: snyk-token   # The name of the GSM Secret
```
Comment (navidshaikh) [inline infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods-externalsecrets.yaml:25]: /lgtm

thanks @spiffxp !

**Review by navidshaikh — APPROVED** [review]

**Comment by spiffxp** [comment]
Moving to ExternalSecrets ref: https://github.com/kubernetes/k8s.io/issues/2220

**Comment by spiffxp** [comment]
We could have caught this in a presubmit I think... `snyk_token` is an invalid secret name, we found this out at deploy time

Fixed in followup https://github.com/kubernetes/k8s.io/pull/2227

**Pull request merged**